### ISSUE TITLE
Show result count in source filter

### ISSUE THREAD

**Issue by ravila4** [biothings/mygeneset.info-website#41]
We could show the number of results on the checkbox list of sources. This is an enhancement that could help users understand how the information is distributed across sources.

![Screenshot from 2022-11-16 07-55-36](https://user-images.githubusercontent.com/9020496/202232735-0ee8cdcc-b6dc-4384-a2c7-765db0183446.png)

This would require be adding `*&aggs=source` to every geneset search query.

https://mygeneset.info/v1/query?q=*&aggs=source.

On a side note, I don't know if the checkbox list is being dynamically generated on the frontend. If a new source is added, will it automatically show up here, and how would the layout accommodate it?

**Comment by vincerubinetti** [comment]
Not too difficult to add I think, but I'll mark as enhancement along with the other feedback from Casey. Once we get other user feedback, I can do these small enhancements in a batch.

> On a side note, I don't know if the checkbox list is being dynamically generated on the frontend. If a new source is added, will it automatically show up here, and how would the layout accommodate it?

The checklist is being generated from the curated sources listed in `/metadata`, so if that updates, so will the checklist.

**Comment by cgreene** [comment]
I think this is related - documentation for how to grab all wikipathways, for example:
https://mastodon.social/@rmflight/109387955847675692

**Comment by rmflight** [comment]
Yeah, I think it would fit @cgreene . As someone who does a fair amount of gene annotation enrichments (GSEA, hypergeometric, etc), having a single source to grab collections of up-to-date genesets would be awesome. 

It seems like it should be doable, especially based on the API docs, but I'm not quite seeing it at the moment.

**Comment by newgene** [comment]
@rmflight Using the universal `biothings_client` package (for any BioThings API) with this new MyGeneset.info API could be handy:

```
import biothings_client
mgs = biothings_client.get_client(biothing_type='geneset', url='https://mygeneset.info/v1')
for geneset in mgs.query('source:wikipathways', fetch_all=True):
    print(geneset['name'])
```

This `fetch_all` parameter returns a long list of query hits as a convenient generator.

Right now, there is a type issue in the `biothing_type` field of https://mygeneset.info/v1/metadata output, otherwise, this should work with just the `url` parameter:

```
mgs = biothings_client.get_client(url='https://mygeneset.info/v1')
```

or even better, we should release a new version of `biothings_client` soon with the `geneset` as a built-in type, then this should work:

```
mgs = biothings.client.get_client('geneset')
```

**Comment by vincerubinetti** [comment]
We're talking about two separate issues in this issue, so I've renamed it. The original issue is a simple one of just adding counts next to some checkboxes from info that can be returned from the API. This is relatively small and trivial.

The other larger issue is talking about downloading _sets_ of genesets; i.e. meta-genesets kind of. I didn't really design the UI with this in mind, I'll have to think about it. Also not sure if the API will allow for us to do this conveniently, I'll look into it.

I guess how I would do it is... in any place where genesets are listed (sets of genesets), I'd have the same download/export section as I have on the geneset editing/viewing page:
<img width="543" alt="Screenshot 2022-11-22 at 4 54 57 PM" src="https://user-images.githubusercontent.com/8326331/203429188-9e9412f2-e244-4a21-be58-f54eb16fa501.png">
And then, as long as I can get all the info from the API, export them all as individual CSV/TSV/JSON/GMX files, and zip them together for the user to download.

Assuming the API supports this, that can be a stop gap for advanced users wanting to do this.

**Comment by rmflight** [comment]
If this isn't part of the design of the site, that's fine. Or if it should only be available via the JSON api, that's also fine. However, having a single place to get access to these various gene sets, with common gene identifiers across them, seems like a really excellent idea, regardless of use case.

Although, if the goal is not really to support gene sets for full on GSEA / hypergeometric enrichment, but just finding genes by keyword and what else is in that geneset, that *feels* like a niche use case, but I'm also a bioinformatician, not a biologist. :wink:

**Comment by vincerubinetti** [comment]
I think the original was supposed to just be a way to collect genes together into sets, manage them (just having them in one place by creating your own account), sharing them, and exporting/downloading them. I think the use case you bring up -- let's call it meta-genesets to avoid confusion -- is a valid one. Right now the search functionality allows you to filter genesets, which sort of checks the "manage a meta-geneset". We could put the active search filters into the URL, which would allow "sharing of a meta-geneset". The last big gap, which you're alluding to, is being able to "export a meta-geneset". It's certainly doable from the frontend, it'll just take some thought.

**Comment by newgene** [comment]
@vincerubinetti as another UI idea for this feature, we could also create a `biothings_client`-based code dump of the genesets exporting from the UI as the batch download option. If we assume the need of this feature is only for the informaticians who have no problem running an exported Python script (also to modify to fit the specific need).
